getting_started/guides/godot_essentials_guide.rst

```diff
@@ -0,0 +1,602 @@
+  hierarchy. This keeps your nodes free of hierarchical dependencies and allows
+  you to reuse them in any manner of situations.
+
+  For more information and examples of signals, see the relevant documentation
+  in 
+  :ref:`GDScript <doc_signals>`, :ref:`C# <doc_c_sharp_differences>`,
+  :ref:`VisualScript <doc_getting_started_visual_script>`, and
+  :ref:`NativeScript C++ <doc_gdnative_cpp_example>` as well as how to
+  :ref:`instance scenes via signals <doc_instancing_with_signals>`.
+
+  For more practical information on when to use signals, see the best practices
+  documentation on :ref:`scene organization <doc_scene_organization>`.
+
+Memory with nodes, references, and resources
+--------------------------------------------
+
+Computer science has many topics related to memory that are worth
+researching such as
+`stack vs. heap <https://www.geeksforgeeks.org/stack-vs-heap-memory-allocation>`__,
+`a breadth/depth of knowledge on data structures <https://www.studytonight.com/data-structures/stack-data-structure>`__,
+`algorithm analysis <https://www.studytonight.com/data-structures/aysmptotic-notations>`__, and
+`data-oriented design <https://en.wikipedia.org/wiki/Data-oriented_design>`__.
+But, as a Godot user, the primary memory 
+issue you must learn to handle is memory management.
+
+Godot uses manual memory management. This means that you are
+responsible for allocating and freeing your own memory. Your
+game *has* to clean up memory at some point. You can either
+do it yourself or via
+`garbage collection <https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)>`__.
+If you do it, you have full control over when you free memory (it is
+"deterministic"). If a garbage collector does it, it will stop
+your game at unpredictable points to handle it for you.
+
+.. note::
+
+  You can read the specifics of Godot's memory handling
+  in its :ref:`core types <doc_core_types>` documentation.
+
+You can delete any Godot Object immediately with the
+`:ref:`.free() <class_Object_method_free>`` method. But, low-level memory
+management is a hassle, so Godot provides tools to help you. Most of the
+time, you work with Objects deriving from either
+:ref:`Node <class_Node>` or :ref:`Resource <class_Resource>`. These classes
+have built-in systems to help you manage memory in deterministic ways.
+
+Nodes
+  Nodes, as you've learned, form node tree hierarchies. Whenever you delete a
+  node, it automatically deletes its children beforehand. And when
+  each of those children receive the command to delete themselves, then they
+  too will delete their children beforehand. This recursive process continues
+  until the leaf nodes, with no children, finally delete themselves. The node
+  tree systematically deletes itself from the bottom up!
+
+  Nodes in particular also have a
+  :ref:`queue_free() <class_Node_method_queue_free>` method. This delays
+  their deletion until the next idle frame to reduce any discernible
+  lag.
```

Comment:
Add link to getting started docs about freeing nodes, which give info about `is_instance_valid()` and `is_queued_for_deletion()`, etc.